rules/TypeDeclaration/Rector/FunctionLike/ReturnTypeDeclarationRector.php

```diff
@@ -220,7 +220,7 @@ private function isNullableTypeSubType(Type $currentType, Type $inferedType): bo
             return false;
         }
 
-        return $inferedType->isSubTypeOf($currentType)
+        return $currentType->isSubTypeOf($inferedType)
             ->yes();
```

Comment:
This was the buggy part. Current type can be stricter then resolved, so if it subtype, it should be kept.